### PR TITLE
Chai integration

### LINKED ISSUE #281: Integrate requests to swap Dai to Chai into the oracle
As soon as https://github.com/poanetwork/tokenbridge-contracts/pull/356 is merged, it is necessary to introduce an additional functionality to the oracle to swap the bridge balance in Dai to Chai to earn an interest generated by DSR.

### PR DESCRIPTION
**Pull request by k1rill-fedoseev**
Closes #281 
- Added dedicated worker for calling `convertToChai` method in `erc-to-native` bridge mode

### CODE REVIEW

File: oracle/docker-compose-erc-native.yml [hunk 1]
```diff
@@ -81,6 +82,20 @@ services:
     entrypoint: yarn worker:swap-tokens
     networks:
       - net_rabbit_bridge_swap_tokens_worker
+  bridge_convert_to_chai_worker:
+    cpus: 0.1
+    mem_limit: 500m
+    build:
+      context: ..
+      dockerfile: oracle/Dockerfile
```
Comment (patitonar) [inline oracle/docker-compose-erc-native.yml]: Please update this branch with the latest changes from `develop` to include the changes from https://github.com/poanetwork/tokenbridge/pull/280

The `build` tag should be replaced with the image from docker hub.
```yaml
image: poanetwork/tokenbridge-oracle:latest
```

Comment (k1rill-fedoseev) [reply oracle/docker-compose-erc-native.yml]: Done

File: oracle/config/transfer-watcher.config.js [hunk 2]
```diff
@@ -29,6 +29,11 @@ if (!transferWatcherRequired) {
   process.exit(EXIT_CODES.WATCHER_NOT_REQUIRED)
 }
 
+const workerQueueConfig = {}
+if (baseConfig.id === 'erc-native') {
+  workerQueueConfig.workerQueue = 'convert-to-chai'
```
Comment (akolotov) [inline oracle/config/transfer-watcher.config.js]: I cannot find where it is being used... Could you help me?

Comment (k1rill-fedoseev) [reply oracle/config/transfer-watcher.config.js]: It is used in `transfer-watcher` for `ERC_TO_NATIVE` mode.
When the corresponding transfer event is caught, a message is sent to `convert-to-chai` queue, so that a validator will check if there is a need to call `convertDaiToChai`.

File: oracle/src/workers/convertToChai.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      })
+
+      logger.debug({ gasEstimate }, 'Gas estimated')
+    } catch (e) {
+      if (e instanceof HttpListProviderError) {
+        const errorMsg = 'RPC Connection Error: convertToChai Gas Estimate cannot be obtained.'
+        logger.error(e, errorMsg)
+        throw new Error(errorMsg)
+      } else {
+        if (chaiContract === null) {
```
[the remark this answers is not in the document]
Comment (k1rill-fedoseev) [reply oracle/src/workers/convertToChai.js]: Seems that, this piece is not needed anymore. I will remove it.

File: oracle/config/transfer-watcher.config.js (hunk 2 above)
Comment (akolotov) [reply oracle/config/transfer-watcher.config.js]: Ok. I see just now. Quite unexpected for me that the watcher did not reported with an error about absence of `config.workerQueue` previously.

Comment (patitonar) [reply oracle/config/transfer-watcher.config.js]: Yes, this config usage is a little hard to follow. It was built having in mind that not all watchers have a side worker to which push jobs beside of the sender. The first one was `half-duplex-watcher` to push jobs to the `swap-tokens` worker. So only if `workerQueue` is defined in the config, it will try to push a job to that queue, that's why there is no error present.

https://github.com/poanetwork/tokenbridge/blob/bc6dd131939f5995d3e1ee95843026ad9ddf5134/oracle/src/services/amqpClient.js#L26-L28

https://github.com/poanetwork/tokenbridge/blob/bc6dd131939f5995d3e1ee95843026ad9ddf5134/oracle/src/watcher.js#L189-L191

I hope this helps to clarify how it works

Comment (akolotov) [reply oracle/config/transfer-watcher.config.js]: Thanks

Comment (akolotov) [reply oracle/config/transfer-watcher.config.js]: Consider to revisit logic in `transfer-watcher` to avoid sending of messages to the worker queue if the contracts are still not updated (has no support of Chai token at all) or earning interest with Chai is deactivated.
